src/dashboardUsers/controller/edit-userByAdmin.js

```diff
@@ -1,48 +1,54 @@
-require("../../authentication/controller/passport-config");
-
 const mongoose = require("mongoose");
 const bcrypt = require("bcrypt");
 const UserModel = require("../../authentication/model/user-model");
 
 exports.updateUser = () => async (req, res, next) => {
   {
     const hashConst = 10;
-    const obj = {};
+    const { _id } = req.body;
+    const userData = req.body;
 
-    const validationId = mongoose.Types.ObjectId.isValid(req.body._id);
+    const validationId = mongoose.Types.ObjectId.isValid(_id);
     if (!validationId) {
       return res.status(400).json({
         error: "Sorry, but provided id is inccorect!",
         data: [],
       });
     } else {
-      const keys = ["firstname", "lastname", "phone", "role", "password"];
-
-      for (let itemsFromBodyIndex in req.body) {
-        if (keys.includes(itemsFromBodyIndex)) {
-          if (itemsFromBodyIndex == "password") {
-            const passwordHash = await bcrypt.hash(
-              itemsFromBodyIndex,
-              hashConst
-            );
-            obj[itemsFromBodyIndex] = passwordHash;
+      const foundUser = await UserModel.findOne({ _id });
+      if (!foundUser) {
+        return res.status(400).json({
+          error: "Sorry, this user doesn't exist in the database!",
+          data: [],
+        });
+      } else {
+// Here we have a conditional block that checks if the edited mail exists in the database and if the password has been changed it will be encrypted and sent like this.
+        
+        if (userData.password || userData.email) {
+          if(userData.password && userData.email){
```

Comment:
It seems that the logic for updating the password and the mail is duplicated within the conditionals.

Reply:
Yes, you're right, I've changed.

Reply:
It still doesn't seem that this works if just a password change is intended (or just an email change is intended).